### PR TITLE
Update findFormItem so it can return checked radio buttons

### PR DESCRIPTION
**Pull request by clguiman**
This is useful when you want to get the value of a radio button in the validate method of f:validateButton

### CODE REVIEW

File: war/src/main/webapp/scripts/hudson-behavior.js [hunk 1]
```diff
@@ -350,7 +350,14 @@ function findNext(src,filter) {
 
 function findFormItem(src,name,directionF) {
     var name2 = "_."+name; // handles <textbox field="..." /> notation silently
-    return directionF(src,function(e){ return (e.tagName=="INPUT" || e.tagName=="TEXTAREA" || e.tagName=="SELECT") && (e.name==name || e.name==name2); });
+    return directionF(src,function(e){ 
+        if (e.tagName == "INPUT" && e.type=="radio" && e.checked==true) {
```
Comment (oleg-nenashev) [inline war/src/main/webapp/scripts/hudson-behavior.js]: And no name in the condition?

File: war/src/main/webapp/scripts/hudson-behavior.js [hunk 2]
```diff
@@ -350,7 +350,14 @@ function findNext(src,filter) {
...
+            var r = 0;
+            while (e.name.substring(r,r+8)=='removeme') //radio buttons have must be unique in repeatable blocks so name is prefixed
```
Comment (oleg-nenashev) [inline war/src/main/webapp/scripts/hudson-behavior.js]: I doubt it is the most effective method to find the value within the string

Comment (clguimanMSFT) [inline war/src/main/webapp/scripts/hudson-behavior.js]: I wanted to use the same code as in buildFormTree: https://github.com/jenkinsci/jenkins/blob/b4095bc4b6c62023a2029e5e2faef8ad0e3a4252/war/src/main/webapp/scripts/hudson-behavior.js#L2564

File: war/src/main/webapp/scripts/hudson-behavior.js (hunk 1 above)
Comment (clguimanMSFT) [inline war/src/main/webapp/scripts/hudson-behavior.js]: the name will not match here, I need to parse out the 'removeme'. I'm doing that inside the if

**Review by oleg-nenashev — APPROVED** [review]
LGTM. A test would be useful on the other hand

**Comment by clguimanMSFT** [comment]
I'm new to the project, can you please point me to where I should add my tests? I wasn't aware there were js unit tests

**Comment by oleg-nenashev** [comment]
@clguimanMSFT In this case it is rather a subject for Jenkins Acceptance Test Harness: https://github.com/jenkinsci/acceptance-test-harness/ .

**Comment by oleg-nenashev** [comment]
@jenkinsci/code-reviewers Would appreciate additional +1 from a JS guru

**Comment by oleg-nenashev** [comment]
@jenkinsci/code-reviewers Ping. @daniel-beck Are you fine with the merge tomorrow?

**Pull request merged**